src/node/hooks.h

```diff
@@ -76,4 +76,33 @@ namespace ccf
       }
     }
   };
+
+  class NetworkConfigurationsHook : public kv::ConsensusHook
```

Comment:
I'm not sure I understand why we have this parallel hook to update the consensus configuration. Could we remove the existing hook on the `nodes.info` table altogether if we're planning to use this new table going forward?

Reply:
In theory, yes. For now my plan was to keep both of them to avoid LTS issues. Keeping just one table would definitely make all of this easier.

Reply:
Is that right? I think the learner/trusted and retiring/retired distinction remains important, so unless I'm missing something we'll continue to need both?

Reply:
Yes, that's right, we need the same amount of information, but we could put all of it into one table instead of two.